feat(replays): add unmask alert to replay details

michellewzhang opened the pull request:
Add a banner on the replay video to surface how to unmask/unblock non-sensitive information. This replaces the dom alert banner.
![SCR-20240926-bgbe](https://github.com/user-attachments/assets/7e46e0e7-e874-41ed-bf61-f928d8495d7c)

The Learn More link goes to https://docs.sentry.io/platforms/javascript/session-replay/privacy/#privacy-configuration.

- This messaging is only shown if the replay is configured to mask all text, mask all input, and block all media (aka all the default options). These are read from `replay.sdkOptions`
- This messaging is only shown the first few 3 times a user views a replay. After the 3rd time, this banner does not show.
- A user can dismiss the banner, in which case the banner will not show unless local storage is cleared.


closes https://github.com/getsentry/team-replay/issues/479

jas-kas commented:
Looks great Michelle!

How about the Learn More link going to "https://docs.sentry.io/platforms/javascript/session-replay/privacy/#privacy-configuration" since it's closer to the SDK config option table?

michellewzhang commented:
> https://docs.sentry.io/platforms/javascript/session-replay/privacy/#privacy-configuration

i have updated! also changed the link for the configure replay card since that one was also linking to the top

ryan953 commented on static/app/components/replays/unmaskAlert.tsx:
```diff
@@ -1,40 +1,52 @@
+const LOCAL_STORAGE_KEY = 'replay-unmask-alert-dismissed';
 
-function PlayerDOMAlert() {
+function unmaskAlert() {
```
feels like this should be uppercase because it's a component.

michellewzhang replied:
oh, good catch

ryan953 commented on static/app/components/replays/unmaskAlert.tsx:
```diff
@@ -44,7 +56,7 @@ const DOMAlertContainer = styled('div')`
 `;
 
-const DOMAlert = styled('div')`
+const UnmaskAlert = styled('div')`
```
ah, naming collision. internal names like this could be less verbose, `Alert` or `Layout` kind of a name.

michellewzhang replied:
i just changed this to `Alert`!

ryan953 commented on static/app/components/replays/unmaskAlert.tsx:
```diff
@@ -1,40 +1,52 @@
+  const {data, isError, isPending} = useUserViewedReplays();
 
-  if (isDismissed) {
+  if (isDismissed || isError || isPending || (data && data.data.length > 3)) {
```
so they'll view 3 replays within the timeframe, after that we'll hide the banner. but when the timeframe restarts they'll see it again...

they can dismiss it forever though (via localstorage).

michellewzhang replied:
i didn't specify a timeframe in the query, so the timeframe is whatever the default is in the endpoint. but that brings up a good point - is it weird for the banner to come back after a period of time?
